client/boot/project/wordpress-com.js

```diff
@@ -85,7 +87,11 @@ export function setupMiddlewares( currentUser, reduxStore ) {
 	// Render Layout only for non-isomorphic sections.
 	// Isomorphic sections will take care of rendering their Layout last themselves.
 	if ( ! document.getElementById( 'primary' ) ) {
-		renderLayout( reduxStore );
+		page( '*', function( context, next ) {
```

Comment:
I updated this. Would love your thoughts on the new implementation.

The problem is that on multi-tree pages that are not server-rendered, we need to render `Layout` before the section tries to render itself. Otherwise, when it tries to do so, the `#primary` and `#secondary` elements are not there for it to render itself into.

This wasn't a problem before when `renderLayout` was synchronous; we just rendered it directly inside of `setupMiddlewares`. But now that we're rendering the layout asynchronously, I think we need to do it in middleware to defer the call of `next()` until after the Layout is rendered.

The new middleware does so conditionally now, so it won't re-render every time.

cc @nb @gziolo